File: TransCoder_tokenized_test_set_functions/python/REARRANGE_ARRAY_ARRI.py
def fix ( A, len ) :
    for i in range ( 0, len ) :
        if ( A [ i ] != - 1 and A [ i ] != i ) :
            x = A [ i ];
            
            while ( A [ x ] != - 1 and A [ x ] != x ) :
                y = A [ x ]
                A [ x ] = x
                x = y
            A [ x ] = x;
            
            if ( A [ i ] != i ) :
                A [ i ] = - 1;
                
    

File: TransCoder_tokenized_test_set_functions/cpp/REARRANGE_ARRAY_ARRI.cpp
int fix ( int A [ ], int len ) {
  for ( int i = 0;
  i < len;
  i ++ ) {
    if ( A [ i ] != - 1 && A [ i ] != i ) {
      int x = A [ i ];
      while ( A [ x ] != - 1 && A [ x ] != x ) {
        int y = A [ x ];
        A [ x ] = x;
        x = y;
      }
      A [ x ] = x;
      if ( A [ i ] != i ) {
        A [ i ] = - 1;
      }
    }
  }
}